mirgecom/operators.py

```diff
@@ -0,0 +1,71 @@
+r""":mod:`mirgecom.operators` provides helper functions for composing DG operators.
+
+.. autofunction:: jump
+.. autofunction:: dg_div
+"""
+
+__copyright__ = """
+Copyright (C) 2021 University of Illinois Board of Trustees
+"""
+
+__license__ = """
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+"""
+
+
+# placeholder awaits resolution on grudge PR #71
+def jump(trace_pair):
+    r"""Return the "jump" in the quantities represented by the *trace_pair*.
+
+    The jump in a quantity $\mathbf{q}$ is denoted $[\mathbf{q}]$ and is
+    defined by:
+    .. math:
+        [\mathbf{q}] = \mathbf{q}^+ - \mathbf{q}^-
+
+    Parameters
+    ----------
+    trace_pair: :class:`grudge.sym.TracePair`
+    Represents the quantity for which the jump is to be calculated.
+
+    Returns
+    -------
+    like(trace_pair.int)
+    """
+    return trace_pair.ext - trace_pair.int
+
+
+def dg_div(discr, vol_flux, bnd_flux):
```

Comment:
Eh, to be honest, I'm not sure this really helps much. I sort of like seeing all the DG-related constructions self-contained in the Euler routines.

Reply:
I do see where you are coming from - especially as a DG person.  I like this wrapper better though because it allows me to write what I see as an uncluttered expression that looks very much like the equation I am trying to implement, and without having to worry about dealing with the sign conventions.

Reply:
Sure, I can live with this if you feel it's very helpful